### DISCUSSION TITLE
[@types/react-dom] 18.0.10 doesn't allow for proper typing with UMD imports

### DISCUSSION THREAD

**Discussion by henry-filosa** [DefinitelyTyped/DefinitelyTyped#64112] [Issues with a @types package]
Our team uses UMD imports. In the react-dom UMD package the functions of `react-dom/client` are in the main bundle.
The correct import is therefor `import { createRoot, Root } from "react-dom";`

However, these are exposed in a separate index file in the types library which requires  "react-dom/client" for Typescript to pick up these exports. So the import has to be `import { createRoot, Root } from "react-dom/client";` to satisfy Typescript, but this won't work at runtime.

Is it possible to adjust the typing library for the UMD case? I'm also trying to figure out if the [Typescript config](https://www.typescriptlang.org/tsconfig#module) can be used to redirect the proper import to the typing file.

[This is similar to this issue, but it seems those developers were not using UMD imports.](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/43848)

[a comment by an automated account is not shown]

**Comment by henry-filosa** [comment]
[We use the dojo loader](https://dojotoolkit.org/reference-guide/1.10/loader/amd.html) which is pretty legacy, so this might just be on us. If anyone else is in this rare boat, you can fix the issue by importing `import { createRoot, Root } from "react-dom/client"` and amending the dojo package loader to [map `react-dom/client` to `react-dom`.](https://dojotoolkit.org/reference-guide/1.10/loader/amd.html#map-config)

```
map: {
        "*": {
            "react-dom/client": "react-dom"
        }
    }
```

**Comment by jordanbtucker** [comment]
I created https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67934 for this.
